app/src/androidTest/java/com/github/h3lp3rs/h3lp/testutils/ViewWaiter.kt

```diff
@@ -0,0 +1,45 @@
+package com.github.h3lp3rs.h3lp.testutils
+
+import android.view.View
+import androidx.test.espresso.PerformException
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.util.HumanReadables
+import androidx.test.espresso.util.TreeIterables
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers.any
+import java.util.concurrent.TimeoutException
+
+/**
+ * Utility test class to wait for a view to appear in case of asynchronous events
+ * TODO: This has not yet been tested

```

Comment:
Still not tested as not yet needed (I thought it could help the swipe problem but it did not :().
I did not want to remove it though as it may be needed later on, and this way we don't have to look it up again. But I do agree that putting it on main may be silly (sadly we don't have a dev branch). I can remove it, but the team should definitely not start from scratch on this in that case.

Reply:
Regarding the contentDescription , these were not my files. I may not be the best person to define the most appropriate description and don't want to step on their toes.
Adding a more common listener like a double tap may solve the issue indeed (and adding a button **will** do the trick), but is it worth it just to make Cirrus happy? Like I said, the code plummeting our coverage right now is the custom listener. However it makes a lot of sense to test this with unit tests! But no point in doing this until we can make that count as coverage.

I hope this addresses your comments! Thanks a lot for taking the time for this not so funny PR (the other one should be better) :)